modules/vba_documents/app/workers/vba_documents/upload_processor.rb

```diff
@@ -154,19 +155,28 @@ def perfect_metadata(parts, upload, timestamp)
       metadata['source'] = "#{upload.consumer_name} via VA API"
       metadata['receiveDt'] = timestamp.in_time_zone('US/Central').strftime('%Y-%m-%d %H:%M:%S')
       metadata['uuid'] = upload.guid
+      check_size(parts[DOC_PART_NAME])
       doc_info = get_hash_and_pages(parts[DOC_PART_NAME])
       metadata['hashV'] = doc_info[:hash]
       metadata['numberPages'] = doc_info[:pages]
       attachment_names = parts.keys.select { |k| k.match(/attachment\d+/) }
       metadata['numberAttachments'] = attachment_names.size
       attachment_names.each_with_index do |att, i|
         att_info = get_hash_and_pages(parts[att])
+        check_size(parts[att])
         metadata["ahash#{i + 1}"] = att_info[:hash]
         metadata["numberPages#{i + 1}"] = att_info[:pages]
       end
       metadata
     end
 
+    def check_size(file_path)
+      if File.size(file_path) > MAX_PART_SIZE
+        raise VBADocuments::UploadError.new(code: 'DOC106',
+                                            detail: 'Max part size exceeded. Limit is 100MB.')
```

Comment:
Can you revise text slightly: 'Maximum document size exceeded. Limit is 100MB per document.'